frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sModel.ts

```diff
@@ -0,0 +1,30 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector } from 'react-redux';
+import { UseK8sModel } from '../../../extensions/console-types';
+import { getGroupVersionKindForReference, transformGroupVersionKindToReference } from '../k8s-ref';
+
+/**
+ * Hook that retrieves the k8s model for provided K8sGroupVersionKind from redux.
+ * @param groupVersionKind K8sGroupVersionKind should be provided
+ * @return An array with the first item as k8s model and second item as inFlight status
+ * @example
+ * ```ts
+ * const Component: React.FC = () => {
+ *   const [model, inFlight] = useK8sModel({ group: 'app'; version: 'v1'; kind: 'Deployment' });
+ *   return ...
+ * }
+ * ```
+ */
+export const useK8sModel: UseK8sModel = (k8sGroupVersionKind) => {
```

Comment:
Any reason this is not exported as the default?

Reply:
nothing specific, this was already a named export and in code I saw most of our hooks are exposed as a named export, so I followed that. LMK if there are any conventions for hooks will move it to default export.

Reply:
Personally I like seeing default export for any file that is explicitly named. If it's a collection file, no default export. Hook or Component, it's just the Pascal vs Camel case, nothing to do with default exports imo. I'm okay with this as-is though. it's not a big deal.